### PR TITLE
Service Fabric Reliable Volume yaml files

### PR DESCRIPTION
**Pull request by ashishnegi**
## Purpose

* Users need yaml files for deploying app targeting service fabric reliable volume. This PR adds sample yaml counter app backed by reliable volume.
* Visual studio integration of sf reliable volume yaml files. Complete deployment log [here](https://gist.github.com/ashishnegi/e8f4d4c29bda010c844a2ae0b73088b9#file-deployment-visual-studio-sfreliablevolume-txt)

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe: `Templates and Visual studio integration`.
```

## How to Test
*  Get the code
```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```
* cd ` src/counter/src/counterSFVolume/windows/App Resources`
```
* az group create --name testrg1 --location eastus && az mesh deployment create --resource-
group testrg1 --input-yaml-files .\app.yaml,.\network.yaml,.\gateway.yaml  --parameters "{'location' : {'value' : 'eastus'}}"
```

* Or open counter.sln file in visual studio.
* Build and publish.

* Test the code

Visit the counter app at the gateway endpoint that you get after the deployment.

## What to Check
Verify that the following are valid
* You can see a webpage with a counter incrementing every second.

### CODE REVIEW

File: src/counter/src/azurefiles/counter/App Resources/network.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-          applicationName: counterApp
-          serviceName: counterService
+## Network definition ##
+network:
+  schemaVersion: 1.0.0-preview2
+  name: counterAppNetwork
+  properties:
+    description: Azure Service Fabric Mesh Counter Application network.
+    addressPrefix: 10.0.0.4/22
+    ingressConfig:
```
Comment (raunakpandya) [inline src/counter/src/azurefiles/counter/App Resources/network.yaml]: The network yaml format in preview2 is different. We need to fix that and add Gateway yaml if not added.

Comment (ashishnegi) [reply src/counter/src/azurefiles/counter/App Resources/network.yaml]: i have moved back to `preview1` for azurefiles as it requires more changes that @wangcai0124  is working on.
Lets have separate PR for `azurefiles` with all fixes.

**Comment by ashishnegi** [comment]
ping @raunakpandya / @anantshankar17 / @mevora-msft

File: src/counter/src/azurefiles/counterService/Service Resources/service.yaml [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
+                - name: ASPNETCORE_URLS
+                  value: http://+:8080
+              resources:
+                requests:
+                  cpu: 0.5
+                  memoryInGB: 0.5
+          replicaCount: 1
+          networkRefs:
+            - name: counterAppNetwork
+              endpointRefs:
```
Comment (anantshankar17) [inline src/counter/src/azurefiles/counterService/Service Resources/service.yaml]: endpointsRef format is for the preview2 schema, not require in preview1

File: src/counter/src/sfvolume/counter/linux/App Resources/app.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+application:
+  schemaVersion: 1.0.0-preview2
+  name: counterApp
+  properties:
+    description: Azure Service Fabric Mesh Counter Application.
+    services:
```
[the remark this answers is not in the document]
Comment (anantshankar17) [reply src/counter/src/sfvolume/counter/linux/App Resources/app.yaml]: Cool

File: src/counter/src/azurefiles/counterService/Service Resources/service.yaml (hunk 2 above)
Comment (ashishnegi) [reply src/counter/src/azurefiles/counterService/Service Resources/service.yaml]: Thanks for catching this.

Comment (ashishnegi) [reply src/counter/src/azurefiles/counterService/Service Resources/service.yaml]: Thanks for catching this.

**Pull request merged**